### PR TITLE
#29 - Fix memory issue in Grf2Fst2 for graphs containing more than 50…

### PR DESCRIPTION
**Pull request by adminviavoo**
… states

Removing the hardcoded size of 50 for int pointers in loop checker of Grf2Fst2.
Tests replayed OK, with memory tests.

### CODE REVIEW

File: Grf2Fst2_lib.cpp [hunk 1]
```diff
@@ -1678,13 +1678,12 @@ for (i=0;i<grf->n_states;i++) {
                 grf->states[next_box]->has_loop = vector_int_contains(grf->states[next_box]->transitions,next_box);
                 grf->states[next_box]->is_first = 1;
                 if (grf->states[next_box]->has_loop != 1) {
-                    //int *visited = NULL;
-                    int visited[50];
+                    // Assigning with twice the potential max size of states
+                    int *visited = (int*)malloc(2*grf->n_states*sizeof(int));
```
Comment (martinec) [inline Grf2Fst2_lib.cpp]: It is required to check whether the `malloc()` was successful or not:

```cpp
if (visited == NULL) {
  fatal_alloc_error("compile_grf");
}
```

Comment (adminviavoo) [inline Grf2Fst2_lib.cpp]: ok

File: Grf2Fst2_lib.cpp [hunk 2]
```diff
@@ -1636,8 +1636,8 @@ for (i=0;i<grf->n_states;i++) {
                 } else if(input[j] == ':' && input[j-1] != '\\' && in_Token == 0) {
                     found = 1;
                     break;
-                }
```
Comment (adminviavoo) [inline Grf2Fst2_lib.cpp]: ok

**Comment by adminviavoo** [comment]
I performed the requested change, hope it is ok for merge.

**Comment by martinec** [comment]
Thank you!

**Pull request closed without merging**